### ISSUE TITLE
Better branch model

### ISSUE THREAD

**Issue by cyberpro4** [dddomodossola/remi#53]
Actually every changes / fixes / features, are pushed on master branch, but, the master branch should contain only stable / tested release.
Can I suggest [this](http://nvie.com/posts/a-successful-git-branching-model/) model?

**Comment by nzjrs** [comment]
I would rather do FF rebases onto master than merges anyway.

But I think this is all a bit too early as IMO REMI is quite far from a stable API where a branch model has to be considered.

**Comment by dddomodossola** [comment]
Yeah, we should use that git model to ensure stable versions on master and make the repository more readable and well organized.

**Comment by dddomodossola** [comment]
John, what is the purpose of an FF rebase? I have't deep knowledge of git usage.

**Comment by nzjrs** [comment]
For single commits - tiny bugfixes or other small things that are not large
enough to be called a 'feature', it keeps the history more readable as you
don't have a merge commit for every commit. For example, if a user submits
a pull request with one commit then this generates a merge commit also (if
merged using the git web interface). That makes the history seem messy.

That extends for local master merges, where git creates a merge commit
because you ```git pull```. Buy most definitions this is not a feature
branch, it is just that your HEAD didn't agree with master and git created
merge to make them agree.

Just look at the git history for all the recent small fixes by contributors
- do all the merge commits make things readable? not really I think.

**Comment by dddomodossola** [comment]
Thank you for the explanation. You are right, infact I felt 'unconfortable' when submitting the merges of pull requests. However this is another aspect not strictly related with what cyberpro4 says. I mean, both thinks can coexist. We can organize better the repository for the future changes and furthermore use FF rebase. Right?

**Comment by nzjrs** [comment]
Yes, we can, and I do in other projects, use both. FF for small things, and feature branches + merges for large things.
